validator/client/log.go

```diff
@@ -45,16 +45,23 @@ func (v *validator) LogNextDutyTimeLeft(slot uint64) error {
 	}
 
 	var nextDutySlot uint64
-	var role string
+	attestingCount := 0
+	proposingCount := 0
 	for _, duty := range v.duties.CurrentEpochDuties {
-		if duty.AttesterSlot > slot && (nextDutySlot > duty.AttesterSlot || nextDutySlot == 0) {
+		if duty.AttesterSlot == nextDutySlot {
+			attestingCount += 1
+		} else if duty.AttesterSlot > slot && (nextDutySlot > duty.AttesterSlot || nextDutySlot == 0) {
```

Comment:
I think the below is easier to consume, clear separation of concern and dont have to manually set `attestingCount` to 1. What do you think?

```go
if duty.AttesterSlot > slot && (nextDutySlot > duty.AttesterSlot || nextDutySlot == 0) {
nextDutySlot = duty.AttesterSlot
}
if duty.AttesterSlot == nextDutySlot {
attestingCount++
}
```

Reply:
Please check the above for the reason of reseting `attestingCount` and `proposingCount`.

Right now it is a bit mixed because in the single `duties` loop we do 2 things, (1) finding the smallest future `slot` that has duties, (2) counting the duties (`attesting`/`proposing`) of that `slot`. It can be more clear to make them into separate loops, we do (1) first and use the result of (1) to do (2), but it will be less efficient as we loop `duties` twice.